core/cap-0035.md

```diff
@@ -97,7 +97,7 @@ flags. This operation can be used to unset `TRUSTLINE_CLAWBACK_ENABLED_FLAG`.
 
 This patch of XDR changes is based on the XDR files in tag/commit `v15.1.0` (`90b2780584c6390207bf09291212d606896ce9f8`) of [stellar-core].
 
-```diff
+```diff mddiffcheck.ignore=true

```

Comment:
I would have liked to fix this CAP-35 diff, but I can't figure out what is corrupted. It's not just whitespace, and even using the reject mode in git apply wasn't particularly useful.

Reply:
Let's try to figure out that problem before merging this otherwise things are going to get frustrating very fast and people are going to leave this disabled.

I agree there is probably a bug in mddiffcheck - maybe using `patch` directly would help (`git apply` may do a bunch of things we don't care about)

Reply:
```sh
$ git patch
git: 'patch' is not a git command. See 'git --help'.
```
What do you mean use patch?

I don't think we should use patches, that's a different thing to diffs, and the CAP process calls for diffs, not patches. Patches include all the commit data, not just the content change, and would bloat a CAP.

---

>I agree there is probably a bug in mddiffcheck

Okay, so it looks like this is a line endings problem. The patch contains windows line ending characters CRLF, and mddiffcheck is preserving them because the markdown parser's AST preserves them, but when I copy the patch via vim on my machine it removes them and applies fine.

A quick check shows that CAP-35 is the only cap that has CRLF line endings. Maybe we should ensure we don't commit CRLFs so as to keep all the CAPs consistent.

Either way, I can probably improve mddiffcheck so it deals with line endings better, but we should also keep CAPs consistent.

Reply:
After a little more digging I'm not sure the line endings issues can be really fixed. Git apply isn't designed to handle line endings discrepancies, and we should just make sure our CAPs are all committed as LF not CRLF.

Reply:
`patch` as in [the unix command](https://linux.die.net/man/1/patch) (not a git subcommand). It can process uni-diffs as a parameter.
Line ending: can't you just normalize line ending in the tool so that `patch` will work? You're already extracting the diff from the md file?

Reply:
I could normalize the line endings to LF, but if we make sure files in the repo are all LF to begin with, this isn't an issue and has other benefits. We can do that with #952 and #956.

Also, mddiffcheck doesn't know if a file should be CRLF or LF, and it'd be a bit weird for it to have that responsibility.|

Reply:
> `patch` as in [the unix command](https://linux.die.net/man/1/patch) (not a git subcommand)

`patch` will work instead of `git apply` because by default it replaces CRLFs with LFs, but the tool is harder to use. We can fix this issue so that both `patch` and `git apply` will work, and this test should make sure that the basic `git apply` should work.

Reply:
agree let's try to enforce encoding and line ending first